sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -339,6 +339,9 @@ private[sql] object PartitioningUtils {
   private val upCastingOrder: Seq[DataType] =
     Seq(NullType, IntegerType, LongType, FloatType, DoubleType, StringType)
 
+  /**
+   * Validate partition columns for writing executions.
+   */
```

Comment:
Since `validatePartitionColumn` is used for only writing-related classes, I added this description for clarification.
We can update the function description if the usage pattern is changed in the future.